pkg/ingester/shipper.go

```diff
@@ -185,42 +187,22 @@ func (s *Shipper) Sync(ctx context.Context) (uploaded int, err error) {
 	return uploaded, nil
 }
 
-// sync uploads the block if not exists in remote storage.
-// TODO(khyatisoneji): Double check if block does not have deletion-mark.json for some reason, otherwise log it or return error.
+// update local meta.json file with Thanos-specific details, and upload the block
+// to blocks storage.
```

Comment:
according to "effective go" the method comments should start with the name of the method:

```
Doc comments work best as complete sentences, which allow a wide variety of automated presentations. The first sentence should be a one-sentence summary that starts with the name being declared.
```

[ref](https://go.dev/doc/effective_go#commentary)

so something like:

```suggestion
// upload updates the local meta.json file with Thanos-specific details, and uploads
// the block to blocks storage.
```